numpy/core/code_generators/generate_umath.py

```diff
@@ -97,6 +97,8 @@ def TD(types, f=None, astype=None, in_=None, out=None, simd=None):
         out = (out,) * len(types)
     elif out is None:
         out = (None,) * len(types)
+    if len(in_) != len(out):
+        raise RuntimeError("Number of inputs don't match outputs")
```

Comment:
Would be better as
```
elif len(out) != len(types):
    raise ValueError(...)
```
and same for the inputs

Reply:
Yeah, that makes far more sense.